client-sdk/ts-web/core/src/client.ts

```diff
@@ -585,18 +594,31 @@ export class GRPCWrapper {
                 // `errorToGrpc` leaves it blank.
                 if (details.length === 1 && details[0].type_url === '' && details[0].value) {
                     const grpcError = misc.fromCBOR(details[0].value) as GRPCError;
-                    if (!e.message) {
-                        e.message = `Message missing, module=${grpcError.module} code=${grpcError.code}`;
-                    }
-                    e.oasisCode = grpcError.code;
-                    e.oasisModule = grpcError.module;
+                    const innerMessage =
+                        e.message ||
+                        `Message missing, module=${grpcError.module} code=${grpcError.code}`;
+                    const message = `callUnary method ${method}: ${innerMessage}
+Invocation stack:
+${invocationStack}
+End of invocation stack`;
+                    // @ts-expect-error options and cause not modeled
+                    const wrapped = new OasisCodedError(message, {cause: e});
+                    wrapped.oasisCode = grpcError.code;
+                    wrapped.oasisModule = grpcError.module;
+                    throw wrapped;
                 }
             }
-            // grpc-web gives us a plain object with no stack trace, which is less helpful.
-            // Transfer it to an Error.
-            const error = new Error(e.message);
-            Object.assign(error, e);
-            throw error;
+            // Just in case there's some non-Error rejection reason that doesn't come with metadata
+            // from oasis-core as expected above, try using JSON to stringify it so that we don't
+            // end up with [object Object].
+            const innerMessage = e instanceof Error ? e.toString() : JSON.stringify(e);
+            const message = `callUnary method ${method}: ${innerMessage}
+Invocation stack:
+${invocationStack}
+End of invocation stack`;
+            // @ts-expect-error options and cause not modeled
+            const wrapped = new Error(message, {cause: e});
+            throw wrapped;
```

Comment:
Output in Chrome:

error.toString()

- after:
  ```
  Error: callUnary method https://grpc.oasis.dev/oasis-core.Staking/Account: RpcError: Http response at 400 or 500 level
  Invocation stack
  at NodeInternal.callUnary (http://localhost:3000/static/js/vendors~main.chunk.js:3840:29)
  at NodeInternal.stakingAccount (http://localhost:3000/static/js/vendors~main.chunk.js:4242:17)
  ...40 lines
  End of invocation stack
  ```

- original error:
  ```
  RpcError: Http response at 400 or 500 level
  ```

error.stack

- after:
  ```
  Error: callUnary method https://grpc.oasis.dev/oasis-core.Staking/Account: RpcError: Http response at 400 or 500 level
  Invocation stack
  at NodeInternal.callUnary (http://localhost:3000/static/js/vendors~main.chunk.js:3840:29)
  at NodeInternal.stakingAccount (http://localhost:3000/static/js/vendors~main.chunk.js:4242:17)
  ...40 lines
  End of invocation stack
    at http://localhost:3000/static/js/vendors~main.chunk.js:3874:23
  ```

- original error:
  ```
  Error: Http response at 400 or 500 level
    at new E (index.js:21)
    at Y (index.js:56)
    at oc.<anonymous> (index.js:55)
    at Ib (index.js:33)
    at O (index.js:32)
    at wc (index.js:43)
    at yc (index.js:46)
    at oc.push../node_modules/grpc-web/index.js.n.W (index.js:44)
    at oc.push../node_modules/grpc-web/index.js.n.R (index.js:44)
  ```

error.cause?.stack

- after
  ```
  Error: Http response at 400 or 500 level
    at new E (index.js:21)
    at Y (index.js:56)
    at oc.<anonymous> (index.js:55)
    at Ib (index.js:33)
    at O (index.js:32)
    at wc (index.js:43)
    at yc (index.js:46)
    at oc.push../node_modules/grpc-web/index.js.n.W (index.js:44)
    at oc.push../node_modules/grpc-web/index.js.n.R (index.js:44)
  ```

Reply:
I don't like how much different this behaves than normal errors. I'd prefer
```ts
const wrapped = new Error(`callUnary method ${method}: ${e.message}`, { cause: e });
wrapped.stack += `\nInvocation stack:\n${invocationStack}\nEnd of invocation stack`;
throw wrapped;
```

wrapped.toString()
```
Error: callUnary method https://grpc.oasis.dev/oasis-core.Staking/Account: RpcError: Http response at 400 or 500 level
```

wrapped.stack
```
Error: callUnary method https://grpc.oasis.dev/oasis-core.Staking/Account: RpcError: Http response at 400 or 500 level
    at http://localhost:3000/static/js/vendors~main.chunk.js:3868:23
Invocation stack:
Error
    at NodeInternal.callUnary (http://localhost:3000/static/js/vendors~main.chunk.js:3840:29)
    at NodeInternal.stakingAccount (http://localhost:3000/static/js/vendors~main.chunk.js:4237:17)
    ...40 lines
End of invocation stack
```

Reply:
I saw that .stack is non-standard. Reading it and potentially getting undefined is one thing. But should we really try setting it?

Reply:
or
```ts
const wrapped = new Error(`callUnary method ${method}: ${e.message}`, { cause: e });
wrapped.stack += `\nCause stack:\n${cause.stack}\nEnd of cause stack`;
wrapped.stack += `\nInvocation stack:\n${invocationStack}\nEnd of invocation stack`;
throw wrapped;
```

Reply:
:shrug:

Reply:
https://github.com/lukaw3d/run-in-everything/tree/f37fdfddbf9c4d6573cbcc1766fd54c43d046ad6 even webkit doesn't break

Reply:
ok let's go with that